lib/minicbor-lease/src/lib.rs

```diff
@@ -0,0 +1,153 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+//! An adapter implementing [`minicbor::encode::write::Write`] for
+//! [`idol_runtime::Leased`] byte buffers.
+
+#![no_std]
+
+/// An adapter implementing [`minicbor::encode::write::Write`] for
+/// [`idol_runtime::Leased`] byte buffers.
+pub struct LeasedWriter<'lease, A>
+where
+    A: idol_runtime::AttributeWrite,
+{
+    lease: &'lease mut idol_runtime::Leased<A, [u8]>,
+    pos: usize,
+    ran_out_of_space: bool,
+}
+
+/// Errors returned by [`LeasedWriter::check_err`].
+#[derive(Copy, Clone, PartialEq)]
+pub enum Error {
+    /// The other side of the lease has gone away.
+    WentAway,
+    /// Data could not be written as there was no room left in the lease.
+    EndOfLease,
+}
+
+/// Errors returned by the [`minicbor::encode::write::Write`] implementation for
+/// [`LeasedWriter`].
+#[derive(Copy, Clone, PartialEq)]
+pub struct WriteError(());
```

Comment:
Any reason not to do
```suggestion
pub struct WriteError;
```
?